scripts/staking_payouts.js

```diff
@@ -0,0 +1,367 @@
+        ({ reward: a }, { reward: b }) => (a != null ? a.add(b) : b),
+        { reward: null }
+      )}`
+    );
+  }
+
+  const payoutTxs$ = from(rewards).pipe(
+    mergeMap(([stashAddress, eras]) =>
+      from(eras.map(assoc("stashAddress", stashAddress)))
+    ),
+    mapRx(({ stashAddress, era }) =>
+      api.tx.staking.payoutStakers(stashAddress, era)
+    ),
+    batchExtrinsics(api, BatchSize),
+    signAndSendPayouts(api, txSender)
+  );
+
+  await lastValueFrom(
+    payoutTxs$.pipe(
+      tap(({ hash, tx }) => {
+        console.log(` * Batch transaction finalized at block \`${hash}\`: [`);
+        for (const arg of tx.method.args[0]) {
+          console.log(
+            `   \`${arg.get("args").validator_stash}\` rewarded for era ${
+              arg.get("args").era
+            }`
+          );
+        }
+        console.log(" ]");
+      })
+    )
+  );
+}
+
+/**
+ * @typedef ErasInfo
+ * @prop {Array<BN>} eras
+ * @prop {Object<string, *>} pointsByEra
+ * @prop {Object<string, *>} rewardsByEra
+ * @prop {Object<string, *>} prefsByEra
+ */
+
+/**
+ * Fetches information for the recent eras with given limit.
+ *
+ * @param {*} api
+ * @returns {ErasInfo}
+ */
+const fetchErasInfo = async (api) => {
+  const eras = await api.derive.staking.erasHistoric();
+  const indexByEra = indexBy(o((era) => era.toString(), prop("era")));
+
+  const [pointsByEra, rewardsByEra, prefsByEra] = await Promise.all([
+    api.derive.staking._erasPoints(eras),
+    api.derive.staking._erasRewards(eras),
+    api.derive.staking._erasPrefs(eras),
+  ]).then(map(indexByEra));
+
+  return {
+    eras,
+    pointsByEra,
+    rewardsByEra,
+    prefsByEra,
+  };
+};
+
+/**
+ * @typedef StashEras
+ * @prop {Array<{ era: BN, eraReward: BN, reward: BN }>} eras
+ * @prop {string} stashAddress
+ */
+
+/**
+ * Retuns era rewards need to be paid grouped by stash ids.
+ *
+ * @param {*} api
+ * @param {ErasInfo} erasInfo
+ * @param {*} validators
+ * @returns {Array<StashEras>}
+ */
+const getUnclaimedStashEras = (api, { eras }, validators) => {
+  const singleStashEras = async (api, stashAddress, allEras) => {
+    const ledger = (
+      await api.query.staking.ledger(stashAddress)
+    ).unwrapOrDefault();
+
+    return differenceWith((a, b) => a.eq(b), allEras, ledger.claimedRewards);
+  };
+
+  return from(validators).pipe(
+    mergeMap(
+      (stashAddress) =>
+        from(singleStashEras(api, stashAddress, eras)).pipe(
+          filterRx(complement(isEmpty)),
+          mapRx(assoc("eras", __, { stashAddress }))
+        ),
+      ConcurrentRequestsLimit
+    )
+  );
+};
+
+/**
+ * Signs and sends extrinsics produced by the given observable.
+ *
+ * @param {*} api
+ * @param {*} initiator
+ * @param {Rx.Observable<import("@polkadot/types/interfaces").Extrinsic>} payouts$
+ *
+ * @returns {Rx.Observable<string>} executed tx with hashes
+ */
+const signAndSendPayouts = curry((api, initiator, payouts$) => {
+  const nonces$ = timer(0, 1000).pipe(
```

Comment:
I think you are trying to get few nonces, 1 for each batch and then send. But can you please document the sections appropriately.